src/app/spans.rs

```diff
@@ -0,0 +1,106 @@
+use crate::trace_context;
+use linkerd2_error::Error;
+use opencensus_proto::trace::v1 as oc;
+use std::collections::HashMap;
+use std::{error, fmt};
+use tokio::sync::mpsc;
+
+const SPAN_KIND_SERVER: i32 = 1;
+const SPAN_KIND_CLIENT: i32 = 2;
+
+#[derive(Clone)]
+pub struct SpanConverter {
+    kind: i32,
+    sink: mpsc::Sender<oc::Span>,
+    labels: HashMap<String, String>,
+}
+
+#[derive(Debug)]
+pub struct IdLengthError {
+    id: Vec<u8>,
+    expected_size: usize,
+    actual_size: usize,
+}
+
+impl error::Error for IdLengthError {
+    fn description(&self) -> &str {
```

Comment:
Nit: `Error::description` is [soft deprecated](https://doc.rust-lang.org/std/error/trait.Error.html#method.description), implementing it shouldn't be necessary.